src/main/java/com/okta/sdk/framework/ApiClient.java

```diff
@@ -64,9 +61,20 @@
     ////////////////////////////////////////////
 
     public ApiClient(ApiClientConfiguration config) {
+        Proxy proxy = ProxySelector.getDefault().select(URI.create(config.getBaseUrl())).iterator().next();
+        URI proxyUri = URI.create(proxy.type() + "://" + proxy.address());
```

Comment:
Okay. That's a good question. Here's the Javadoc for [**ProxySelector#select(URI)**](https://docs.oracle.com/javase/8/docs/api/java/net/ProxySelector.html#select-java.net.URI-):

> **select**
>
> public abstract List<Proxy> select(URI uri)
> Selects all the applicable proxies based on the protocol to access the resource with and a destination address to access the resource at. The format of the URI is defined as follow:
>
>  * http URI for http connections
>  * https URI for https connections
>  * socket://host:port for tcp client sockets connections
>
> **Parameters:**
> uri - The URI that a connection is required to
> 
> **Returns:**
> a List of Proxies. Each element in the the List is of type Proxy; when no proxy is available, the list will contain one element of type Proxy that represents a direct connection.
>
> **Throws:**
> IllegalArgumentException - if the argument is null

I think we're covered there. In the case where there is no proxy it will return a single element list with a DIRECT proxy.